### DISCUSSION TITLE
Manage DC, CS, RST, BL pins with an IO Expander

### DISCUSSION THREAD

**Discussion by michcfr** [moononournation/Arduino_GFX#167] [General]
Hello,
My board (based on the ESP32) do not manage directly the DC, CS, RST, BL pins but uses an I2C expander. So the activation of these pins (LOW, HIGH) is a C function call to the I2C expander chip.
Looking at the source code of Arduino_GFX, I noticed the functions:
DC_LOW()
DC_HIGH()
CS_LOW()
CS_HIGH()

Is there an elegant way to override these functions with my own code: myDC_LOW(), myDC_HIGH(), myCS_LOW(), myCS_HIGH() ?

I did not noticed the equivalent with RST and BL but I can imagine I can create our own function:
myRST_LOW()
myRST_HIGH()

myBL_HIGH()
myBL_LOW()

Regards,
Michel

**Comment by moononournation** [comment]
You can connect RST and BL in expander but I don't think connect DC and CS to expander is a good idea. In contrast, reserve one GPIO for DC and direct pull down the CS will be much better.

**Comment by michcfr** [comment]
Can you please elaborate why it is not a good idea?
FYI, I already connected DC and CS and to expander using the TFT_eSPI library, and it works very well. Now I want to switch to Arduino_GFX because it can drive many displays at a time contrary to the TFT_eSPI library.

**Reply by moononournation** [reply]
If you accept the performance, you can implement your own databus class inherent from Arduino_Databus

**Comment by michcfr** [comment]
ok
Sorry, but I don't see how and where I can manage the code to get HIGH or LOC the DC, CS, RST, BL pins in Arduino_Databus? any help please?

**Reply by moononournation** [reply]
e.g. copy Arduino_HWSPI

**Reply by michcfr** [reply]
thx. now it is clear
